pyccel/parser/semantic.py

```diff
@@ -2656,6 +2659,12 @@ def _visit_FunctionDef(self, expr, **settings):
                             args_inout[i] = True
 
                         i_fa += 1
+                if isinstance(a, Variable):
+                    if a.is_const and (args_inout[i] or (str(a) in all_assigned)):
+                        msg = 'Argument ({}) must be an inout!'.format(a)
```

Comment:
Sorry for this late comment, but I think that we should give a more meaningful error message to the user.

A non-Fortran programmer may not understand what `intent` means, so I recommend
```python
msg = "Cannot modify 'const' argument ({})".format(a)
```